pkg/git/strategy/strategy_proxy_test.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategy
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/elazarl/goproxy"
+	"github.com/fluxcd/pkg/gittestserver"
+	. "github.com/onsi/gomega"
+
+	"github.com/fluxcd/source-controller/pkg/git"
+	"github.com/fluxcd/source-controller/pkg/git/gogit"
+	"github.com/fluxcd/source-controller/pkg/git/libgit2"
+)
+
+// These tests are run in a different _test.go file because go-git uses the ProxyFromEnvironment function of the net/http package
+// which caches the Proxy settings, hence not including other tests in the same file ensures a clean proxy setup for the tests to run.
+func TestCheckoutStrategyForImplementation_Proxied(t *testing.T) {
+	proxyAddr := "localhost:9999"
+
+	type testCase struct {
+		name          string
+		gitImpl       git.Implementation
+		url           string
+		branch        string
+		setupGitProxy func(g *WithT, proxy *goproxy.ProxyHttpServer, proxyGotRequest *bool) (*git.AuthOptions, func())
+		usedProxy     bool
+		getError      bool
```

Comment:
Conventionally, we use terms like `wantErr` for when we want error to happen throughout the code base. For `usedProxy`, let's use `wantUsedProxy` in the same fashion.